docs/guides/typescript.md

```diff
@@ -1,50 +1,66 @@
 # TypeScript Support
 
-In this guide, we will go through the steps to get a Hardhat project working with TypeScript. This means that you can write your Hardhat config, tasks, scripts and tests in [TypeScript](https://www.typescriptlang.org/). For a general overview of using Hardhat refer to the [Getting started guide](../getting-started).
+In this guide, we will go through the steps to get a Hardhat project working with TypeScript. This means that you can write your Hardhat config, tasks, scripts and tests in [TypeScript](https://www.typescriptlang.org/). 
 
-## Installing dependencies
+For a general overview of using Hardhat refer to the [Getting started guide](../getting-started).
 
-Hardhat detects if `typescript` and `ts-node` are installed in its npm project,
-and automatically enables TypeScript support.
+## Enabling TypeScript support
 
-To install them, open your terminal, go to your Hardhat project, and run:
+Hardhat will automatically enable its TypeScript support if your config file is written
+in TypeScript, but this requires a few changes to work properly.
+
+### Installing dependencies
+
+Hardhat uses TypeScript and `ts-node` under the hood, so you need to install them.
+To do it, open your terminal, go to your Hardhat project, and run:
 
 ```
 npm install --save-dev ts-node typescript
 ```
 
-You also need these packages:
+To be able to write your tests in TypeScript, you also need these packages:
 
 ```
 npm install --save-dev chai @types/node @types/mocha @types/chai
 ```
 
-## Configuration
+### TypeScript configuration
+
+You can easily turn a JavaScript Hardhat config file into a TypeScript one. Let's see how this is done starting with a
+fresh Hardhat project.
 
-Let's get started with a fresh Hardhat project. Run `npx hardhat` and go through the steps to create a sample project. When you're done your project directory should look like this:
+Open your terminal, go to an empty folder, and run `npx hardhat` and go through the steps to create a sample
```

Comment:
Open your terminal, go to an empty folder, run `npx hardhat` and go through the steps to create a sample